### PR TITLE
[9.0] Allow public upload when the quota is unlimited

### PR DESCRIPTION
**Pull request by nickvergessen**
Backport of https://github.com/owncloud/core/pull/24964
Approval in https://github.com/owncloud/core/pull/24964#issuecomment-223292394

@PVince81 @mmattel

### CODE REVIEW

File: lib/private/helper.php [hunk 1]
```diff
@@ -500,7 +500,7 @@ public static function maxUploadFilesize($dir, $freeSpace = null) {
 	 */
 	public static function freeSpace($dir) {
 		$freeSpace = \OC\Files\Filesystem::free_space($dir);
-		if ($freeSpace !== \OCP\Files\FileInfo::SPACE_UNKNOWN) {
+		if ($freeSpace < \OCP\Files\FileInfo::SPACE_UNLIMITED) {
```
Comment (nickvergessen) [inline lib/private/helper.php]: this one is missing in master, will do a second pr

Comment (nickvergessen) [inline lib/private/helper.php]: https://github.com/owncloud/core/pull/24989

**Comment by PVince81** [comment]
Makes sense :+1:

**Comment by PVince81** [comment]
@owncloud/filesystem

**Comment by mmattel** [comment]
:+1:

**Pull request merged**